js/tji-datasets-explore.js

```diff
@@ -244,74 +249,81 @@ TJIGroupByDoughnutChart.prototype.create_legend = function() {
 
 
 var TJIChartView = function(props){
+
+  var that = this;
+
   this.state = {
-    chart_data: null,
+    active_dataset_index: null,
     filtered_record_indices: null,
-    complete_data: null,
     active_filters: [],
   }
 
+  this.datasets = props.datasets;
+
   this.ui = {
-    $charts: jQuery(props.charts_elt_selector),
-    $filters: jQuery(props.filters_elt_selector),
+    $chartview_charts: jQuery(props.charts_elt_selector),
+    $charts: null,
+    $chartview_filters: jQuery(props.filters_elt_selector),
     $form: null,
+    $summary: null,
+    $description: null,
+    $select_dataset: null,
     $download: null,
     $record_count: null,
+    $loader: null,
   }
 
   this.templates = {
     chart_wrapper_template: props.chart_wrapper_template,
-    chartview_description_template: props.chartview_description_template,
+    chartview_charts_template: props.chartview_charts_template,
+    chartview_summary_template: props.chartview_summary_template,
   }
 
   this.components = {
     charts: [],
     autocompletes: [],
   }
 
-  this.chart_configs = props.chart_configs;
-  this.filter_configs = props.filter_configs;
+  this.create_chartview_DOM();
+
+  this.set_active_dataset(0);
 
-  this.get_data(props.compressed_data_json_url, props.complete_data_csv_url);
+  this.datasets[this.state.active_dataset_index].fetch_data
```

Comment:
we only want to attach_events once, since we're not destroying any of the DOM elements when we change state. and the summary looked weird rendering before all the data was present.